### ISSUE TITLE
use of aws-signing-proxy

### ISSUE THREAD

**Issue by AnamikaN** [chef-customers/aws_native_chef_server#73]
If we dont want to use aws-signing-proxy, what will be changes , we need to make to the template? Basically I am more interested in understanding the role of aws-signing-proxy. can anyone share their thoughts please?

**Comment by irvingpop** [comment]
I can't find the previous conversation about why we need this, so I'll start over :) 

tl;dr - the [aws-signing-proxy](https://github.com/chef-customers/aws-signing-proxy) service is what chef-server uses to communicate with AWS ES. Don't take this out, or you will suffer outages.

1. In the beginning, AWS ES didn't support VPCs - all traffic to AWS ES had to be signed, or whitelisted by IP 
2. Originally we used the IP whitelist method, requiring a NAT gateway.  that really limited flexibility and introduced security concerns.
3. Later on we flipped over to using a bit of Nginx LUA code to sign requests, that was more secure and let us eliminate the NAT gateway requirement
4. We then learned that customers were experiencing outages, because the AWS ES IPs were changing but nginx would not pick up the changes without a restart.  Note: Even without the Nginx LUA proxy, chef server uses Nginx to proxy all requests to Elasticsearch.
5.  After trying many workarounds in Nginx (none reliably worked), we adopted the aws-signing-proxy because it was simple, easy to understand and met all our operational requirements.

**Comment by AnamikaN** [comment]
Thank you so much for your response. Really appreciate it.
